src/Umbraco.Core/Services/Implement/MediaService.cs

```diff
@@ -553,6 +577,16 @@ private IQuery<IMedia> GetPagedDescendantQuery(string mediaPath)
 
             return _mediaRepository.GetPage(query, pageIndex, pageSize, out totalChildren, filter, ordering);
         }
+        private IEnumerable<IMedia> GetPagedLocked(IQuery<IMedia> query, long pageIndex, int pageSize, 
```

Comment:
This code can be shared in one method which can accept a bool as to whether or not to include the count